client/src/components/ShelfEditor.js

```diff
@@ -0,0 +1,214 @@
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { toast } from 'react-toastify';
+import { useState } from 'react';
+
+import CloseIcon from '../images/closeIcon.svg';
+import SaveAddButton from './SaveAddButton';
+import validateShelfSelection from '../lib/validateShelfSelection';
+
+export default function ShelfEditor({
+  isStatic,
+  shelves,
+  book,
+  onSetIsSelector,
+  onSelectShelf,
+}) {
+  const initialSelection = {
+    bookshelfId: '',
+    columnId: '',
+    compartmentId: '',
+  };
+  const [selection, setSelection] = useState(initialSelection);
+  const [shelfIndex, setShelfIndex] = useState(null);
+  const [columnIndex, setColumnIndex] = useState(null);
+
+  function handleBookShelfChange(event) {
+    const shelfIndex = shelves.findIndex(
+      (shelf) => shelf._id === event.target.value
+    );
+    shelfIndex >= 0 ? setShelfIndex(shelfIndex) : setShelfIndex(null);
+    setSelection({
+      ...selection,
+      bookshelfId: event.target.value,
+      columnId: '',
+      compartmentId: '',
+    });
+    toast.dismiss('validationError');
+  }
+
+  function handleColumnChange(event) {
+    const columnIndex = shelves[shelfIndex].columns.findIndex(
+      (column) => column._id === event.target.value
+    );
+    columnIndex >= 0 ? setColumnIndex(columnIndex) : setColumnIndex(null);
+    setSelection({
+      ...selection,
+      columnId: event.target.value,
+      compartment: '',
+    });
+    toast.dismiss('validationError');
+  }
+
+  function handleCompartmentChange(event) {
+    setSelection({ ...selection, compartmentId: event.target.value });
+    toast.dismiss('validationError');
+  }
+
+  function handleSelectionSave(event) {
+    event.preventDefault();
+    if (validateShelfSelection(selection)) {
+      onSelectShelf(selection, book);
+      onSetIsSelector(false);
+      toast.success('Book added to Shelf!', {
+        position: toast.POSITION.BOTTOM_CENTER,
```

Comment:
for me, this worked with simply typing ```position: 'bottom-center'``` or is there a certain reason you did it this way? (: